crates/wit-component/src/encoding.rs

```diff
@@ -2155,7 +2155,7 @@ world test {
 "#,
             )
             .unwrap();
-        let world = resolve.select_world(&pkgs, None).unwrap();
+        let world = resolve.select_world(&[pkg], None).unwrap();
```

Comment:
You might already be planning to get to this, but `select_world` can be updated to taking a `PackageId` now instead of a list which will have a ripple effect throughout this repository.